include/mc_rbdyn/Robot.h

```diff
@@ -1032,9 +1032,26 @@ struct MC_RBDYN_DLLAPI Robot
   /** Copy loaded data from this robot to a new robot **/
   void copyLoadedData(Robot & destination) const;
 
-  /** Used to set the surfaces' X_b_s correctly and create frames from RobotModule */
+  /**
+   * Used to set all robot surfaces' X_b_s correctly
+   *
+   * This updates all surfaces in surfaces_ vector and should only be called
+   * once
+   */
   void fixSurfaces();
 
+  /**
+   * Used to set the specified surfaces' X_b_s correctly
+   *
+   * This function should only be called once per surface.
+   *
+   * @param surfaces Surfaces that need to be modified
+   **/
+  void fixSurfaces(const std::vector<SurfacePtr> & surfaces);
```

Comment:
I think we can simplify this into `void fixSurface(Surface & surface)` instead, we only need to call `fixSurfaces` after:
- loading from an rsdf folder, this can be done as the surfaces are being copied into the robot
- after copying the surfaces from another robot, this also can be done one by one

Reply:
Done in https://github.com/jrl-umi3218/mc_rtc/commit/27508c95af9192905f9fc4031436057c8dd6a01a